src/profile-logic/perf-script-profile-format.js

```diff
@@ -0,0 +1,219 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+/**
+ * The "perf script" format is the plain text format that is output by an
+ * invocation of `perf script`, where `perf` is the Linux perf command line tool.
+ */
+export function isPerfScriptFormat(profile: string): boolean {
+  const firstLine = profile.substring(0, profile.indexOf('\n'));
+  return /^(\S.+?)\s+(\d+)\/*(\d+)*\s+([\d.]+)/.test(firstLine);
```

Comment:
See my comment below about the regex, I'd also prefer to have a permalink here.

Reply:
That may come from gregg - marcus?

Reply:
@mstange ^

Reply:
I don't think it's taken from Gregg's flame graphs, as there is not a direct match on anything when I did a search. I'm also not sure on the licensing for those regexes, as there is some GPL code in it.

I'm fine taking this on for now, and possibly following up with more docs/tests around individual regexes as needed if we touch it.